x-pack/plugins/index_management/public/lib/render_badges.js

```diff
@@ -6,15 +6,26 @@
 
 
 import React, { Fragment } from 'react';
-import { EuiBadge } from '@elastic/eui';
+import { EuiBadge, EuiSearchBar } from '@elastic/eui';
 import { getBadgeExtensions } from '../index_management_extensions';
-export const renderBadges = (index) => {
+export const renderBadges = (index, filterChanged) => {
   const badgeLabels = [];
-  getBadgeExtensions().forEach(({ matchIndex, label, color }) => {
+  getBadgeExtensions().forEach(({ matchIndex, label, color, filterExpression }) => {
     if (matchIndex(index)) {
+      const clickHandler = () => {
+        filterChanged
+          && filterExpression
+          && filterChanged(EuiSearchBar.Query.parse(filterExpression));
+      };
       badgeLabels.push(
         <Fragment key={label}>
-          {' '}<EuiBadge color={color}>{label}</EuiBadge>
+          {' '}
+          <EuiBadge
+            color={color}
+            onClick={clickHandler}
```

Comment:
How would you feel about adding some more instructions for screen readers? I wonder if it's obvious that selecting this button would filter on it.

```jsx
aria-label={i18n.translate(
  'key',
  {
    defaultMessage: '{label}. Select to filter on this.'
    values: { label },
  },
)}
```

Reply:
Yup makes sense to me.  Pushed this change.